zilliqa/src/sync.rs

```diff
@@ -161,12 +168,24 @@ impl Sync {
             .sync
             .max_blocks_in_flight
             .clamp(max_batch_size, Self::MAX_BATCH_SIZE);
-
+        let sync_base_height = config.sync.base_height;
+        let prune_interval = config.sync.prune_interval;
         // Start from reset, or continue sync
         let latest_block_number = latest_block
             .as_ref()
             .map_or_else(|| u64::MIN, |b| b.number());
 
+        // If set, sync_base_height must be sane
+        if sync_base_height != u64::MAX && latest_block_number < sync_base_height {
+            return Err(anyhow::anyhow!("sync_base_height > highest_block"));
+        }
+
+        let zq1_ceil_height = config
+            .consensus
+            .get_forks()?
+            .find_height_fork_first_activated(crate::cfg::ForkName::ExecutableBlocks)
```

Comment:
As discussed on Slack, I think this would be cleaner with a call to something like `forks.get(block.number).is_executable` when you receive a block.